Milestone 5: Final feature

PhillipUg opened the pull request:
### In this milestone i have done the following:
- converted App functional component into class
- added `handleClick` function to app component
- passed `handleClick` function down into ButtonPanel then into Button as a prop

bellom commented on src/logic/calculate.js:
```diff
@@ -2,31 +2,42 @@ import operate from './operate';
 const calculate = (data, btn) => {
   let { total, next, operation } = data;
-  const operators = ['+', '&times;', '&minus;', '÷'];
+  const operators = ['+', 'x', '-', '÷', '%'];
```
**we need to work on the operator `÷ `**
Division by zero is not handled properly yet. Let's fix this.
The reason is that when a division by zero happens, the app crashes.

Steps to reproduce:

Divide a number by zero (e.g. 4 / 0)
The app crashes.

bellom commented on src/logic/operate.js:
```diff
@@ -6,9 +6,9 @@ const operate = (num1, num2, operation) => {
-    case '&minus;':
+    case '-':
       return numb1.minus(numb2).toString();
     case '%':
```
**we need to work on the operator `% `**
the operator seems not working.

When it comes to `%` we only need one operand.
example: click `10` then `%` = 0.01

example: `num1 = (num1 *= 0.01).toString();`

bellom commented on src/components/App.js:
```diff
@@ -1,15 +1,33 @@
+    this.state = {
+      total: null,
+      next: null,
+      operation: null, /*eslint-disable-line*/
```
we are indirectly changing the rules by disabling this rule, let's fix the codebase instead.

bellom reviewed (changes requested):
## Hi @PhillipUg  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 
 
**Kindly check out my comments below;**


### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

PhillipUg commented:
### I have corrected the following:
- handled division by zero
- corrected the `%` operator
- fixed codebase and removed rule disabler

Forison reviewed (changes requested):
## Status:: Change required❌

# 🙋‍♂️ @PhillipUg 

### You have done a good job so far by the issues raised earlier have not been fully addressed🤗.

* [ ] Now the application does not throw into error when a number is divided by zero but does not return the desired results.
- When the number is divided zero, the result should be```undefined```, currently it returns zero which not ideal, kindly address this issue.

* [ ] In relation to the second request in the previous review, When it comes to ```%``` we only need one operand. 8%===0.08 which is good but when we have ie (8-5)% it returns 0.08 which is not right. Let address this flaw. When a user selects 8 then, - then 5 then % the result should 0.03.
The point here is that, when next, operation and total are occupied and the subsequent selections is an operand the current value of next, operation and total must be computed and the new operand should be the new operator value. 

* [ ] Your application should allow continuous computation.
 ie 10-7+3+2 would be handled like below.
10-7 => the next operand is + upon selecting should trigger a computation and return 3 + then
3+3=> next operand is + then upon selecting it should trigger a computation and return 6 + then 
and so on.
> In summary when next, operator and total are occupied and the subsequent selection is an operand, resolve them and store the result in total, let the newly selected operand occupy the operator and then continue

**[OPTIONAL]**

* [ ] I think making the currently selected operand visible make the UI more user friendly

### That is it kindly make the change and submit the project for a re-review👨‍🎨.
#### Happy coding👨‍💻 
Addo Forison

PhillipUg commented:
#### I have corrected the following:
- continuous computation
- returns `undefined` on division by zero
- corrected issue with %
- Display operator between turns

Torres-ssf reviewed (approved):
Milestone approved :+1: 
Hi @PhillipUg :hand: 
Great job implementing the requested changes, your project it's approved.
You may merge the PR now.
Good luck with the next project.

Happy Coding :computer: 
**Sergio Torres**
